docker/development/Dockerfile

```diff
@@ -172,6 +174,37 @@ ARG VERSION
 ENV VERSION="${VERSION}"
 LABEL VERSION="${VERSION}"
 
-# Execution
+# Development image for humans (non-root user)
+
+FROM root-user as non-root-user
+
+# Install dev utilities
+
+RUN apt-get update \
+ && apt-get install -y zsh sudo \
+ && rm -rf /var/lib/apt/lists/*
+
+## Create non-root user and group
+
+ARG USERNAME="vscode"
+ARG USER_UID="1000"
+ARG USER_GID=${USER_UID}
+RUN groupadd --gid ${USER_GID} ${USERNAME} || true \
+ && adduser --uid ${USER_UID} --gid ${USER_GID} ${USERNAME} \
+ && echo "${USERNAME} ALL=(ALL) NOPASSWD: ALL" >> /etc/sudoers.d/${USERNAME}
+
+## Use non-root user
+
+USER ${USERNAME}
+
+## Install shell utilities
+
+RUN sh -c "$(curl -fsSL https://raw.githubusercontent.com/ohmyzsh/ohmyzsh/master/tools/install.sh)" \
+ && git clone https://github.com/bhilburn/powerlevel9k.git /home/${USERNAME}/.oh-my-zsh/custom/themes/powerlevel9k \
+ && git clone https://github.com/zsh-users/zsh-autosuggestions /home/${USERNAME}/.oh-my-zsh/custom/plugins/zsh-autosuggestions \
+ && git clone https://github.com/zsh-users/zsh-syntax-highlighting.git /home/${USERNAME}/.oh-my-zsh/custom/plugins/zsh-syntax-highlighting \
+ && mkdir -p /home/${USERNAME}/.vscode-server/extensions /home/${USERNAME}/.vscode-server-insiders/extensions
+
+## Configure environment
 
-CMD [ "/bin/bash" ]
```

Comment:
Why remove the default hook?

Reply:
I was thinking just because all our make commands specify the shell to be used at runtime. But maybe there is a good reason to keep it.

Reply:
It's nice to have when testing the container build sometimes. You can do `docker run open-space-collective/open-space-toolkit-astrodynamics:latest` without any extra args and it will throw you into a shell. Not required, but I think it's a nice-to-have